packages/salesforcedx-vscode-core/src/commands/forceCreateManifest.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { join } from 'path';
+import { format } from 'util';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import { getRootWorkspacePath } from '../util';
+
+export async function forceCreateManifest(
+  sourceUri: vscode.Uri,
+  uris: vscode.Uri[] | undefined
+) {
+  if (!uris || uris.length < 1) {
+    uris = [];
+    uris.push(sourceUri);
+  }
+  const sourcePaths = uris.map(uri => uri.fsPath);
+  if (sourcePaths) {
+    const componentSet = ComponentSet.fromSource(sourcePaths);
+    const inputOptions = {
+        placeHolder: nls.localize('manifest_editor_save_placeholder'),
+        prompt: nls.localize('manifest_editor_save_prompt'),
+        validateInput: text => {
+          return text.includes('.xml') ? 'Remove file extension' : null;
+        }
+    } as vscode.InputBoxOptions;
+    const response = await vscode.window.showInputBox(inputOptions);
+    if (response === undefined) {
+      // Canceled and declined to name the document
+      openUntitledDocument(componentSet);
+    } else {
+      saveDocument(response, componentSet);
+    }
+  }
+}
+
+function openUntitledDocument(componentSet: ComponentSet) {
+  vscode.workspace.openTextDocument({
+    content: componentSet.getPackageXml(),
+    language: 'xml'
+  }).then(newManifest => {
+    vscode.window.showTextDocument(newManifest);
+  });
+}
+
+function saveDocument(response: string, componentSet: ComponentSet) {
+  const fileName = response ? response.concat('.xml') : 'package.xml';
```

Comment:
One suggestion will be to make the extension optional when a user enters and add it here when missing. But it'll be good to get Ananya's thoughts on this.